pkg/controller/endpoint/endpoints_controller.go

```diff
@@ -221,19 +221,15 @@ func getHostname(pod *v1.Pod) string {
 	if len(pod.Spec.Hostname) > 0 {
 		return pod.Spec.Hostname
 	}
-	if pod.Annotations != nil {
```

Comment:
this can simply become `return pod.Spec.Hostname` or you can fix callers to just read the field directly